### PR TITLE
chore(roll): roll Playwright to 1.42.0-alpha-2024-02-19

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-python/issues/2246
Fixes https://github.com/microsoft/playwright-python/pull/2261

### CODE REVIEW

File: playwright/_impl/_page.py [hunk 1]
```diff
@@ -1029,6 +1046,8 @@ async def pdf(
         preferCSSPageSize: bool = None,
         margin: PdfMargins = None,
         path: Union[str, Path] = None,
+        outline: bool = None,
```
Comment (danphenderson) [inline playwright/_impl/_page.py:1049]: Shouldn't these be `outline: Optional[bool] = None` etc.

**Review by dgozman — APPROVED** [review]

File: playwright/_impl/_page.py (hunk 1 above)
Comment (danphenderson) [reply playwright/_impl/_page.py:1047]: Please let me know if I am missing something.

Comment (mxschmitt) [reply playwright/_impl/_page.py:1047]: We use `strictParameterNoneValue = false` hence changing a single parameter won't make a big difference there.

Comment (mxschmitt) [reply playwright/_impl/_page.py:1047]: https://github.com/microsoft/playwright-python/blob/976f04fb4d56d1497fd85e0c5199a9daeed4a138/pyproject.toml#L44

**Pull request merged**